### DISCUSSION TITLE
Basic Template

### DISCUSSION THREAD

**Discussion by pietraferreira** [SilentVoid13/Templater#685] [Templates Showcase]
This template should:

- Get the name of the file and set it as the title.
- Format the title so that the first letter is capitalized and get rid of any "-" separators.
- Format title metadata to be surrounded by " ".
- Set mouse to the tags section so the first thing I do is set tags.
- Set date of file creation in metadata. 

```js
<%* let title = tp.file.title
  if (title.startsWith("Untitled")) {
    title = await tp.system.prompt("Title");
    await tp.file.rename(title);
  } 
-%>
<%*
  let result = title.replace(/-/g, ' ')
  result = result.charAt(0).toUpperCase() + result.slice(1);
  tR += "---"
%>
title:  <%* tR += "\"" + result + "\"" %>
tags:
<% tp.file.cursor(1) %>
programming-languagues:
created: <% tp.date.now("YYYY-MM-DD") %>
---
# <%* tR += result %>
---
```

For example, a note named "something-i-am-studying.md" would have its title automatically generated as "Something i am studying" in the metadata and "# Something i am studying" in the body.

A nice feature would be to have the first letter of every word capitalized - but I haven't figured that out yet.

**Comment by welpdx** [comment]
> A nice feature would be to have the first letter of every word capitalized
Replace 
```js
<%*
  let result = title.replace(/-/g, ' ')
  result = result.charAt(0).toUpperCase() + result.slice(1);
  tR += "---"
%>
``` 
with

```js
<%*
  let result = title.replace(/\b\w/g, c => c.toUpperCase());
  tR += "---"
%>
```
![img](https://i.imgur.com/b0JrQ5P.jpeg)
Source: google search: [javascript regex capitalize first letter of every word ](https://google.com/search?q=javascript%20regex%20capitalize%20first%20letter%20of%20every%20word%20) -> `replace(/\b\w/g, c => c. toUpperCase());`

[As a noob at regex, I rely alot on google and stackoverflow lol]

~welp

**Reply by rmfberlin** [reply]
I don't know how to but this regex doesn't work. Your code doesn't delete hyphen or underscore and replaces them with a blank. Your code only capitalises the first letter in each word when a hyphen was found, but not after an underscore or something else.

**Reply by AB1908** [reply]
Use regexr.com and play around to find the correct regex

**Reply by SutanuNandigrami** [reply]
Following worked for me:

```js
<%* let title = tp.file.title
  if (title.startsWith("Untitled")) {
    title = await tp.system.prompt("Title");
    await tp.file.rename(title);
  }
-%>
<%*
  let result= title.replace(/[_-]/g, ' ');
  result = result.replace(/\b\w/g, c => c.toUpperCase());
  await tp.file.move("/00 Inbox/Notetaken/" + result);
  tR += "---"
%>
```

**Comment by RDAbreu** [comment]
Thanks! This looks like a good template for my initial incursion into Templater! When I inevitably fail, I promise to at least attempt to report it in a comic manner!
